api-server/src/server/boot/challenge.js

```diff
@@ -355,7 +355,18 @@ export function isValidChallengeCompletion(req, res, next) {
     log('challengeType', challengeType, isNumeric(challengeType));
     return res.status(403).json(isValidChallengeCompletionErrorMsg);
   }
-  if ('solution' in req.body && !isURL(solution)) {
+  // If `backEndProject`:
+  // - `solution` needs to exist, but does not have to be valid URL
+  // - `githubLink` needs to exist and be valid URL
+  if (challengeType === challengeTypes.backEndProject) {
+    if (!solution || !githubLink) {
+      log('isObjectId', id, ObjectID.isValid(id));
+      return res.status(403).json(isValidChallengeCompletionErrorMsg);
+    } else if (!isURL(githubLink)) {
+      log('isObjectId', id, ObjectID.isValid(id));
+      return res.status(403).json(isValidChallengeCompletionErrorMsg);
+    }
```

Comment:
```suggestion
    if (!solution || !isURL(githubLink + '')) {
      log('isObjectId', id, ObjectID.isValid(id));
      return res.status(403).json(isValidChallengeCompletionErrorMsg);
    }
```
This seemed a little verbose.

Reply:
Why do we need the log?

Reply:
No idea. We probably don't.

Reply:
If `!isURL(githubLink + '')` is converting to a string, could we not `!isUrl(String(githubLink))`?

Reply:
I have no idea about the log, but it was in all the others. So, I left it.